packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -76,25 +102,49 @@ export const useStoreSwitcher = props => {
         );
     }, [storeConfigData, availableStoresData]);
 
+    // Get suffix based on page type
+    const getSuffix = useCallback(
+        (storeCode, currentSuffix) => {
+            let suffix = currentSuffix;
+
+            if (pageType === 'CATEGORY') {
+                suffix =
+                    availableStores.get(storeCode).category_url_suffix || '';
+            }
+            if (pageType === 'PRODUCT') {
+                suffix =
+                    availableStores.get(storeCode).product_url_suffix || '';
+            }
+
+            return suffix;
+        },
+        [availableStores, pageType]
+    );
```

Comment:
I see what @eug123 is saying.

This function should return two things - the suffix and the pathname. Because configured suffix delimiter could be a random char, and not `.`, we can't always assume to split on `.`.

In this case, you'll have to use different logic to determine the suffix/pathnames for pdp/category.